src/plugins/ncr/reducers/nodes.js

```diff
@@ -27,6 +28,17 @@ const onPbjxEnvelopeReceived = (prevState, action) => {
   return state;
 };
 
+const onNodesRecieved = (prevState, action) => {
+  const state = { ...prevState };
+  action.nodes.forEach((node) => {
+    const message = node.schema().getCurie().getMessage();
+    state[message] = { ...state[message] };
+    state[message][node.get('_id')] = node;
```

Comment:
@nanetteCodes in the other reducer method we refer to the message as "label", let's do that here too.  also use `state[message][node.get('_id')] = node.freeze();`.  the freeze ensures the node can't be modified.